### PR TITLE
LPS-165007 Add List Type External Reference Code Exception

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://issues.liferay.com/browse/LPS-165007

### CODE REVIEW

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -75,6 +76,16 @@ public ListTypeDefinition addListTypeDefinition(
 			listTypeDefinitionPersistence.create(
 				counterLocalService.increment());
 
+		int countListTypeDefinition =
```
Comment (carolmariaabb) [inline modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java:79]: @mateussgn let's rename it to `count`

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java [hunk 2]
```diff
@@ -75,6 +76,16 @@ public ListTypeDefinition addListTypeDefinition(
...
+			listTypeDefinitionPersistence.countByC_ERC(
+				listTypeDefinition.getCompanyId(), externalReferenceCode);
+
+		if (countListTypeDefinition != 0) {
+			throw new DuplicateListTypeExternalReferenceCodeException(
+				"Duplicated External Reference Code: " +
+				listTypeDefinition.getExternalReferenceCode());
+		}
```
Comment (carolmariaabb) [inline modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeDefinitionLocalServiceImpl.java:87]: @mateussgn let's put this under feature flag

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by mateussgn** [comment]
Thanks for the review, @carolmariaabb! Applying the changes

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:stop

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:reevaluate:1277401_5577

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by carolmariaabb** [comment]
Hey @mateussgn, I think it's safe to send this PR directly to Brian, but please double check with the QA team. Thanks!

**Comment by mateussgn** [comment]
ci:close

**Pull request closed without merging**